### ISSUE TITLE
[FEATURE] Add string split function to PPL

### ISSUE THREAD

**Issue by A-Gray-Cat** [opensearch-project/opensearch-spark#643]
**Is your feature request related to a problem?**
Analysts would want to split a string using a delimiter. For example, when analyzing DNS logs, analysts would split the hostname using the "." delimiter.

**What solution would you like?**
A function that's similar to the split_part(str, delimiter, partNum) function in Spark SQL:
- Splits `str` by delimiter and return requested part of the split (1-based). If any input is null, returns null. if `partNum` is out of range of split parts, returns empty string. If `partNum` is 0, throws an error. If `partNum` is negative, the parts are counted backward from the end of the string. If the `delimiter` is an empty string, the `str` is not split.

**What alternatives have you considered?**

**Do you have any additional context?**

**Comment by YANG-DB** [comment]
@A-Gray-Cat 
this is possible today using the `parse` command as shown [here](https://github.com/opensearch-project/opensearch-spark/blob/fd3f82fc16273d47f17aa47e699aeaa8b4c3679d/ppl-spark-integration/README.md?plain=1#L322)
```
source=accounts | parse address '(?<streetNumber>\d+) (?<street>.+)' | sort num(streetNumber) | fields streetNumber, street
```

**Comment by A-Gray-Cat** [comment]
Yeah. I'm aware of that function. Regex parsing can do what split can do, but for many folks, regex is just difficult to get right in the first run.

Having a dedicated string splitting function can make this experience much easier.

I agree with you that, since the use case is supported by parse, we can prioritize other tasks over this one.
